src/components/screen/SignIn.tsx

```diff
@@ -0,0 +1,345 @@
+import * as AppAuth from 'expo-app-auth';
+import * as Facebook from 'expo-facebook';
+import * as GoogleSignIn from 'expo-google-sign-in';
+
+import { Alert, Platform, ScrollView, TouchableOpacity, View } from 'react-native';
+import { DefaultNavigationProps, User } from '../../types';
+import { Button as DoobooButton, EditText } from '@dooboo-ui/native';
+import React, { ReactElement, useEffect, useState } from 'react';
+import {
+  androidExpoClientId,
+  iOSClientId,
+  iOSExpoClientId,
+} from '../../../config';
+
+import Button from '../shared/Button';
+import Constants from 'expo-constants';
+import { IC_ICON } from '../../utils/Icons';
+import { Ionicons } from '@expo/vector-icons';
+import StatusBar from '../shared/StatusBar';
+import { colors } from '../../theme';
+import { getString } from '../../../STRINGS';
+import styled from 'styled-components/native';
+import { useThemeContext } from '@dooboo-ui/native-theme';
+
+const Container = styled.SafeAreaView`
+  flex: 1;
+  justify-content: center;
+  background: ${({ theme }): string => theme.background};
+`;
+
+const LogoWrapper = styled.View`
+  margin-top: 104px;
+  margin-bottom: 60px;
+`;
+
+const StyledLogoImage = styled.Image`
+  width: 60px;
+  height: 60px;
+`;
+
+const StyledLogoText = styled.Text`
+  color: ${({ theme }): string => theme.fontColor};
+  font-size: 20px;
+  font-weight: bold;
+`;
+
+const Wrapper = styled.View`
+  margin: 0 40px;
+`;
+
+const ButtonWrapper = styled.View`
+  margin-top: 16px;
+  width: 100%;
+  flex-direction: row;
+`;
+
+const FindPwTouchOpacity = styled.TouchableOpacity`
+  padding: 20px;
+  margin-bottom: 16px;
+  align-self: center;
+`;
+
+const FindPwText = styled.Text`
+  color: ${({ theme }): string => theme.tintColor};
+  text-decoration-line: underline;
+`;
+
+const SocialButtonWrapper = styled.View`
+  margin-bottom: 40px;
+`;
+
+interface Props {
+  navigation: DefaultNavigationProps<'SignIn'>;
+}
+
+function SignIn(props: Props): ReactElement {
+  const [isSignIn, setIsSignIn] = useState<boolean>(false);
+  const [signingInFacebook, setSigningInFacebook] = useState<boolean>(false);
+  const [signingInGoogle, setSigningInGoogle] = useState<boolean>(false);
+  const [googleUser, setGoogleUser] = useState<User | null | unknown>(null);
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [errorEmail, setErrorEmail] = useState<string>('');
+  const [errorPassword, setErrorPassword] = useState<string>('');
+  let timer: number;
+
+  const { theme, changeThemeType } = useThemeContext();
+
+  const validateEmail = (email: string): boolean => {
```

Comment:
Since you are also using the same function in `SignUp` screen, I think it'd be better to go in the utility function.